compiler/src/codegen/compcore.re

```diff
@@ -3186,9 +3334,16 @@ let compile_globals = (wasm_mod, env, {globals} as prog) => {
 };
 
 let compile_main = (wasm_mod, env, prog) => {
+  let preamble =
+    if (Config.elide_type_info^ || List.length(prog.type_metadata) == 0) {
+      Expression.nop(wasm_mod);
```

Comment:
No real reason. I actually changed it to that in https://github.com/grain-lang/grain/pull/563/files. It doesn't make much of a difference because Binaryen takes care of it.

Reply:
Sounds good!